src/main/java/org/spongepowered/api/plugin/Plugin.java

```diff
@@ -49,39 +51,43 @@
      * The human readable name of the plugin as to be used in descriptions and
      * similar things.
      *
-     * @return The plugin name
+     * @return The plugin name, or an empty string if unknown
      */
-    String name();
+    String name() default "";
```

Comment:
We can always easily default to an ID if there is no name present. The idea is to make **only** the ID required in the annotation (as identification) so you can specify everything else through optional file metadata.